config/config_mgmt.py

```diff
@@ -232,9 +234,11 @@ def breakOutPort(self, delPorts=list(), addPorts= list(), portJson=dict(), \
             if_name_map, if_oid_map = port_util.get_interface_oid_map(dataBase)
             self.sysLog(syslog.LOG_DEBUG, 'if_name_map {}'.format(if_name_map))
 
-            # If we are here, then get ready to update the Config DB, Update
-            # deletion of Config first, then verify in Asic DB for port deletion,
+            # If we are here, then get ready to update the Config DB,
+            # shutdown the ports, Update deletion of Config,
+            # verify in Asic DB for port deletion,
             # then update addition of ports in config DB.
+            self.shutdownIntf(delPorts)
```

Comment:
Do we need to set admin_status of added ports to UP at line 247? If its done somewhere else, is that right place?

Reply:
It is decided by portJson dict passed to breakout_Ports APIs by Py Click. As I know it is always set to UP.

Reply:
Nope, I dont think so, could you please point me to the line where its set to UP?, below is the line where we create that json.
```
portJson = dict(); portJson['PORT'] = port_dict
```

I dont see where the interface added to port_dict has admin_status set to UP.

Reply:
as Praveen said, it's always set to up when we pass the dictionary.
[portconfig.py#L193](https://github.com/zhenggen-xu/sonic-buildimage/blob/sonic-cfg-mgmt/src/sonic-config-engine/portconfig.py#L193)

Reply:
OK, But I don't see uniformity here. When we call breakOutPort(), we expect it to shutdown ports, delete ports, and add ports. But we are deciding admin_status of newly created ports, outside of this function. OK, I will let Praveen make the call.

Reply:
Since new config for added port is controlled by platform.json and not by configdb.json, we have to pass lanes, speed, index, admin_staus from py click. There is uniformity because PY click just passes it but do not set it to down or up. We set it down and up after config validation in config_mgmt.py only. I feel, this is good,